Sprint 6 05/07/2024

testproEBF opened the pull request:
Added Artists.feature

testproEBF committed:
TestNG and refactoring

Added test for profile name change and positive and negative login test

testproEBF committed:
Homework17

Added positive test for adding song to new playlist

testproEBF committed:
Homework18

Added playing songs using media player buttons.

testproEBF committed:
Homework18-1

Reduced pause time and commented logintest in build.gradle

testproEBF committed:
Homework19

Used parameterization and added delete playlist test

testproEBF committed:
Homework20

Implemented Explicit Waits

testproEBF committed:
Updated Homework20

Removed throws InterruptedException

testproEBF committed:
Homework21

Used actions class in renaming playlist test.

testproEBF committed:
Updated Homework21

Removed previous homework.

testproEBF committed:
Homework22

Used page object model in renaming playlist

testproEBF committed:
Homework23

Implemented Fluent Interfaces (Selenium Page Factory) in renaming playlist

testproEBF committed:
Updated Homewor23

Implemented encapsulation (private and protected) and refactored some methods.

testproEBF committed:
Homework24

Added selenium grid

testproEBF committed:
Update Homework-24

testproEBF committed:
Update Homework-24

testproEBF committed:
Updated Homework24

testproEBF committed:
Updated Homework-24

added some fix for firefox and safari

testproEBF committed:
Homework25

testproEBF committed:
Updated Homework25

testproEBF committed:
Updated Homework25

testproEBF committed:
Updated lamdaTest credentials

testproEBF committed:
Updated Homework25

Made cloud work

testproEBF committed:
Updated Homework25

Resolved issue in renamePlaylist method failed test

testproEBF committed:
Updated Homework25

Cleaned the code

testproEBF committed:
Updated Homework25

testproEBF committed:
Updated Homework25

testproEBF committed:
Updated Homework-25

Removed Thread.wait

testproEBF committed:
Updated Homework27

testproEBF committed:
Updated Homework27

Used cucumber

testproEBF committed:
Updated homework27

testproEBF committed:
Updated Homework29

testproEBF committed:
Updated

testproEBF committed:
updated

testproEBF committed:
Updated

testproEBF committed:
updated

testproEBF committed:
Update LoginPage.java

testproEBF committed:
Update CreateSmartPlaylistStepDefinition.java

testproEBF committed:
Updated

testproEBF committed:
Updated

testproEBF committed:
Updated SprintA

testproEBF committed:
Updated SprintA

Cleaned the code

testproEBF committed:
Updated

testproEBF committed:
Update BaseDefinition.java

testproEBF committed:
Updated Chrome drivers

testproEBF committed:
Updated Login.feature

testproEBF committed:
Added Logout.feature

testproEBF committed:
Updated the Login and Logout features

Updated the email used in email update

testproEBF committed:
Update BasePage.java

testproEBF committed:
Update ProfilePage.java

testproEBF committed:
Cleaned up

testproEBF committed:
Added tests for Current Queue

testproEBF committed:
Updated CurrentQueue page

testproEBF committed:
Added tests in Current Queue

testproEBF committed:
Added Tests in Current Queue

testproEBF committed:
Added tests in Current Queue

testproEBF committed:
Added Current Queue tests

testproEBF committed:
Completed the tests in Current Queue

testproEBF committed:
Updated Login and other pages

Corrected the method for getting the population and implemented AjaxElementLocatorFactory in BasePage and FluentWait for Log in elements.

testproEBF committed:
Cleaned up code

testproEBF committed:
Update

Cleaned up

testproEBF committed:
Added Profile and Preferences feature

testproEBF committed:
Updated

testproEBF committed:
Update build.gradle

testproEBF committed:
Update BaseTest.java

testproEBF committed:
Refactored the runner files

testproEBF committed:
Update

testproEBF committed:
update

testproEBF committed:
Added parallel testing

testproEBF committed:
Updates

testproEBF committed:
Added Artist.feature

testproEBF committed:
updates

Berichicko commented:
Hi Enrile

The code is not compiled! 

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/0b5e87ce-3ed7-49ac-b3a3-0c18d0e6807b)

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/8b062663-a928-4e80-9efc-2c9a879520b3)

**HINT**

**Step 1** 

Add  `testImplementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.17.1'`

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/f3c2e3bc-8b01-4682-b7a3-afa891274753)

```
dependencies {
    testImplementation 'org.testng:testng:7.10.1'
    implementation 'org.seleniumhq.selenium:selenium-java:3.141.59'
    testImplementation 'io.github.bonigarcia:webdrivermanager:5.8.0'
    implementation 'com.opencsv:opencsv:5.7.1'
    implementation 'io.cucumber:cucumber-testng:7.9.0'
    implementation 'io.cucumber:cucumber-core:7.9.0'
    implementation 'io.cucumber:cucumber-java:7.9.0'
    implementation 'io.cucumber:gherkin:25.0.2'
    testImplementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.17.1'
}
```


**Step 2**

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/ad2816ba-23e5-4d72-8f3a-f7ad5e25df73)

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/d5588110-ff3e-4de8-93ff-a3ae9ae7f85f)

**Step 3**

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/f4bf5c8b-0a03-4ff4-8146-67ebe096a221)

**SOLUTION:**

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/44935d8d-679e-482e-b78e-a26ecde78cff)

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/bfc30d32-3017-4ca3-ba56-80684d4cce74)

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/6f0078c3-5ae1-41cd-a244-7100d7d21307)

**Step 4**

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/f0fdb337-f740-4206-9add-88aafa174cab)

**SOLUTION:**

Regarding the hard-coded artist name in the feature file, it's generally better to avoid hard-coding specific data into your feature files whenever possible. Instead, you can use placeholders or variables in your feature file and then pass the actual data through step definitions.

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/166f4d3b-8bee-4ecc-81e7-5c6d4b983664)

```
    @And("I click the {string} artist's thumbnail in {string} view")
    public void clickArtistsThumbnail(String artistName, String viewMode) throws InterruptedException {
        artistsPage.clickArtists(artistName, viewMode);
    }
```

testproEBF commented:
Hi, Alex! I was able to run my code successfully. Here are my results:

![Screenshot 2024-05-07 at 9 10 49 AM](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/791a290d-f21c-4a10-b215-5597cd3cd305)

![Screenshot 2024-05-07 at 9 52 10 AM](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/23a5b2ad-bf5d-4c24-98d6-71ef7c2d8281)

Notes:
1. The results are inconsistent. Sometimes it can scroll down and find the element but sometimes, it does not. I will try your suggestions.
2. I am running my code in JDK 21. When I added the apache dependency, there was a build failure:  Unsupported class file major version 66

<img width="1440" alt="Screenshot 2024-05-07 at 9 57 57 AM" src="https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/0ded6807-64d5-4bad-b590-5a0249e5f1fa">

3. I switched to JDK 22, there is still build failure:  Unsupported class file major version 66

<img width="1440" alt="Screenshot 2024-05-07 at 10 15 46 AM" src="https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/50d27f96-55c4-4718-a4cb-0ffd24c2de31">

Berichicko commented:
> Hi, Alex! I was able to run my code successfully. Here are my results:
> 
> ![Screenshot 2024-05-07 at 9 10 49 AM](https://private-user-images.githubusercontent.com/154494267/328552020-791a290d-f21c-4a10-b215-5597cd3cd305.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTUyMDIwLTc5MWEyOTBkLWYyMWMtNGExMC1iMjE1LTU1OTdjZDNjZDMwNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGZjMWRlOWU5NmVkMjA0YzQ4NGU1ZmExMDc5OTQ3Mzc1MDAyN2U2YWFmMWQ3MjY2NzMyNzMxYTVkNzVjNzJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1h0ZUn84_rl-NpZ0rk4pe-CYJGPyioe30cC8dqFjkcQ)
> 
> ![Screenshot 2024-05-07 at 9 52 10 AM](https://private-user-images.githubusercontent.com/154494267/328552024-23a5b2ad-bf5d-4c24-98d6-71ef7c2d8281.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTUyMDI0LTIzYTViMmFkLWJmNWQtNGMyNC05OGQ2LTcxZWY3YzJkODI4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZGUyZDQ2NjRhYzhkY2E4NTE3NDU3ZmIyNDFkODVlMTZkMDlhMjc5NzFiZDUyYTQ4MTQwZTY4OTE4ZTM5YTI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kKZAfPDX042GAQ7TemgPGhaJ134DXv_slItogJgurjI)
> 
> Notes:
> 
> 1. The results are inconsistent. Sometimes it can scroll down and find the element but sometimes, it does not. I will try your suggestions.
> 2. I am running my code in JDK 21. When I added the apache dependency, there was a build failure:  Unsupported class file major version 66
> 
> <img alt="Screenshot 2024-05-07 at 9 57 57 AM" width="1440" src="https://private-user-images.githubusercontent.com/154494267/328555364-0ded6807-64d5-4bad-b590-5a0249e5f1fa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTU1MzY0LTBkZWQ2ODA3LTY0ZDUtNGJhZC1iNTkwLTVhMDI0OWU1ZjFmYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MTYzOGY4YTNiMzQyNjBhYzUxMDMyMzFhZWQ4YmRkM2U2ZWVlNDQyY2JlZTdjYjlmM2FiZDczNDJiZDFmMDQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kD1BZKeDbdbL37O59F-hZ_VCeYqwL6Tq7lQN_JixMRI">
> 3. I switched to JDK 22, there is still build failure:  Unsupported class file major version 66
> 
> <img alt="Screenshot 2024-05-07 at 10 15 46 AM" width="1440" src="https://private-user-images.githubusercontent.com/154494267/328557765-50d27f96-55c4-4718-a4cb-0ffd24c2de31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTU3NzY1LTUwZDI3Zjk2LTU1YzQtNDcxOC1hNGNiLTBmZmQyNGMyZGUzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNjllZjJhZWE2NGJmYmZmNDczMzBiMzI1MTBiYzZlMzU5ZDAzMGFkZTYzOTY3Y2I4YjQzZGE4NGVlNzI3YzlkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EFWjn04dM5UYlb5bjtGhyM5nKDkIiFqc6WNzTMQqThg">

Hi Enrile 

1) Update dependencies: If you're using any third-party libraries or dependencies, make sure they are compatible with the version of Java you're using. Some libraries may require updates to work with newer Java versions.

Link https://mvnrepository.com/

java  22 supports class 66, but according to [this](https://docs.gradle.org/8.7/userguide/compatibility.html#java) you cannot run gradle 8.7 with java  22.

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/48cec0c5-3577-46f8-9e88-9714566f9a0c)


2) Also You can use Java 11 or Java 17 (https://strawpoll.com/most-popular-java-version)

For example I use Java 11 (https://www.infoworld.com/article/3708994/java-11-and-java-17-lead-java-usage-azul-survey-finds.html)

testproEBF commented:
> **Step 4**
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328502391-f0fdb337-f740-4206-9add-88aafa174cab.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTAzNjUsIm5iZiI6MTcxNTA5MDA2NSwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTAyMzkxLWYwZmRiMzM3LWY3NDAtNDIwNi05YWRkLTg4YWFmYTE3NGNhYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxMzU0MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YmIxZWQ5YmZmNmZhZjliZjk5YzBhMDE1MWYyNWE0YWE5OWVlZjRmYjM1NDA2ZmQwNGQ0ZDZlYmZlMjEzZDM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.V8dlOxQhwpNVglYnXjwyc5ysQOrDCyOyI-OGPncoZs8)
> 
> **SOLUTION:**
> 
> Regarding the hard-coded artist name in the feature file, it's generally better to avoid hard-coding specific data into your feature files whenever possible. Instead, you can use placeholders or variables in your feature file and then pass the actual data through step definitions.
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328506251-166f4d3b-8bee-4ecc-81e7-5c6d4b983664.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTAzNjUsIm5iZiI6MTcxNTA5MDA2NSwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTA2MjUxLTE2NmY0ZDNiLThiZWUtNGVjYy04MWU3LTVjNmQ0Yjk4MzY2NC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxMzU0MjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NzQyMDhkYzEzZDQ3NDMwMjE5YThmMmQ4OGZmNTdmODA1MmUyODI0ZTBlN2YyMDEzMDc0ZmIzNzJhMTk1OTliJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vXtp0qZpKM9bfkFxdJf4x2uT6arortLtpgW0_cdk8fA)
> 
> ```
>     @And("I click the {string} artist's thumbnail in {string} view")
>     public void clickArtistsThumbnail(String artistName, String viewMode) throws InterruptedException {
>         artistsPage.clickArtists(artistName, viewMode);
>     }
> ```

Regarding this one, can you please explain further? What I understand is, there should be placeholders/variables in the feature file and the actual data are passed through step definitions. How do you pass data through step definitions? 

In my Artists.feature file, there are two methods of getting the data:

1. The scenario in feature file has placeholders where the tester can input the test data:

<img width="748" alt="Screenshot 2024-05-07 at 10 26 07 AM" src="https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/55ffcd08-cc1d-4412-8852-ce57baf737fc">

But I change the data in the feature file, too. For example, I can change the "Xylo-Ziko" to "Makaih Beats". I am not changing the data in the step definition.

2. The data is randomly selected from the available data in the UI (example: artists or songs of artist). The tester does not input any data.

<img width="716" alt="Screenshot 2024-05-07 at 10 36 53 AM" src="https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/e76e0977-7660-4085-89ff-91d49f2d90f2">

Additional question: Which of the two methods is usually implemented in the industry?

Thank you!

testproEBF commented:
> > Hi, Alex! I was able to run my code successfully. Here are my results:
> > ![Screenshot 2024-05-07 at 9 10 49 AM](https://private-user-images.githubusercontent.com/154494267/328552020-791a290d-f21c-4a10-b215-5597cd3cd305.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTUyMDIwLTc5MWEyOTBkLWYyMWMtNGExMC1iMjE1LTU1OTdjZDNjZDMwNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMGZjMWRlOWU5NmVkMjA0YzQ4NGU1ZmExMDc5OTQ3Mzc1MDAyN2U2YWFmMWQ3MjY2NzMyNzMxYTVkNzVjNzJkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1h0ZUn84_rl-NpZ0rk4pe-CYJGPyioe30cC8dqFjkcQ)
> > ![Screenshot 2024-05-07 at 9 52 10 AM](https://private-user-images.githubusercontent.com/154494267/328552024-23a5b2ad-bf5d-4c24-98d6-71ef7c2d8281.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTUyMDI0LTIzYTViMmFkLWJmNWQtNGMyNC05OGQ2LTcxZWY3YzJkODI4MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ZGUyZDQ2NjRhYzhkY2E4NTE3NDU3ZmIyNDFkODVlMTZkMDlhMjc5NzFiZDUyYTQ4MTQwZTY4OTE4ZTM5YTI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kKZAfPDX042GAQ7TemgPGhaJ134DXv_slItogJgurjI)
> > Notes:
> > 
> > 1. The results are inconsistent. Sometimes it can scroll down and find the element but sometimes, it does not. I will try your suggestions.
> > 2. I am running my code in JDK 21. When I added the apache dependency, there was a build failure:  Unsupported class file major version 66
> > 
> > <img alt="Screenshot 2024-05-07 at 9 57 57 AM" width="1440" src="https://private-user-images.githubusercontent.com/154494267/328555364-0ded6807-64d5-4bad-b590-5a0249e5f1fa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTU1MzY0LTBkZWQ2ODA3LTY0ZDUtNGJhZC1iNTkwLTVhMDI0OWU1ZjFmYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MTYzOGY4YTNiMzQyNjBhYzUxMDMyMzFhZWQ4YmRkM2U2ZWVlNDQyY2JlZTdjYjlmM2FiZDczNDJiZDFmMDQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.kD1BZKeDbdbL37O59F-hZ_VCeYqwL6Tq7lQN_JixMRI">
> > 3. I switched to JDK 22, there is still build failure:  Unsupported class file major version 66
> > <img alt="Screenshot 2024-05-07 at 10 15 46 AM" width="1440" src="https://private-user-images.githubusercontent.com/154494267/328557765-50d27f96-55c4-4718-a4cb-0ffd24c2de31.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTE3MzQsIm5iZiI6MTcxNTA5MTQzNCwicGF0aCI6Ii8xNTQ0OTQyNjcvMzI4NTU3NzY1LTUwZDI3Zjk2LTU1YzQtNDcxOC1hNGNiLTBmZmQyNGMyZGUzMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDE3MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNjllZjJhZWE2NGJmYmZmNDczMzBiMzI1MTBiYzZlMzU5ZDAzMGFkZTYzOTY3Y2I4YjQzZGE4NGVlNzI3YzlkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EFWjn04dM5UYlb5bjtGhyM5nKDkIiFqc6WNzTMQqThg">
> 
> Hi Enrile
> 
>     1. Update dependencies: If you're using any third-party libraries or dependencies, make sure they are compatible with the version of Java you're using. Some libraries may require updates to work with newer Java versions.
> 
> 
> Link https://mvnrepository.com/
> 
> java 22 supports class 66, but according to [this](https://docs.gradle.org/8.7/userguide/compatibility.html#java) you cannot run gradle 8.7 with java 22.
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328563119-48cec0c5-3577-46f8-9e88-9714566f9a0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTM0NjYsIm5iZiI6MTcxNTA5MzE2NiwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTYzMTE5LTQ4Y2VjMGM1LTM1NzctNDZmOC05ZTg4LTk3MTQ1NjZmOWEwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDQ2MDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZjE2YzY0M2ZkYjE4NjVlMjM5NTM3YzQwYTRjNWE5ZWViYjM0MDFjZTE5ZTQ5NGUzZjE4OTBjZjhmZDM0YjY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.gY4VAQSqXQjGI4ZxD-HhEyThtjzXCXv9oQ82_HsbGck)
> 
>     2. Also You can use Java 11 or Java 17 (https://strawpoll.com/most-popular-java-version)
> 
> 
> For example I use Java 11 (https://www.infoworld.com/article/3708994/java-11-and-java-17-lead-java-usage-azul-survey-finds.html)

Hi! I don't understand what you want me to do here. What I understand is, I should update my dependencies. But all of my dependencies are up to date. Also, I can run my code successfully. There are no problems with running the code on my end. These are my results:
![Screenshot 2024-05-07 at 9 10 49 AM](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/4b3b6445-4005-438c-ac0a-a678aa08f734)
![Screenshot 2024-05-07 at 9 52 10 AM](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/154494267/b870381e-4456-45d7-9f32-61ae1566273d)

Berichicko commented:
> Additional question: Which of the two methods is usually implemented in the industry?
> 
> Thank you!

**Using Fill-in-the-Blanks in Feature Files:** In feature files, we use placeholders like {string} that get filled with real data when we run our tests.
**Putting Data in Action with Step Definitions:** In Java, we have methods called step definitions. They get the real data from feature files and use it to do things in our tests.

**For industry practice:**

- Both ways are common.
- Fill-in-the-blanks keep feature files neat.
- Getting data from the UI can be handy for tests with changing or random data.

Berichicko commented:
You wrote about the error and I wrote you the solution

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/ed02b87d-e7ba-464a-8c95-0c3617093798)

testproEBF commented:
> You wrote about the error and I wrote you the solution
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328571046-ed02b87d-e7ba-464a-8c95-0c3617093798.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTM5MDAsIm5iZiI6MTcxNTA5MzYwMCwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTcxMDQ2LWVkMDJiODdkLWU3YmEtNDY0YS04Yzk1LTBjMzYxNzA5Mzc5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDUzMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMzQ1MjE2ZTU2NmFiNjkwNDE1NjRlMzY1Yjc4Y2ZhM2ZlYjkyNzg4YzMxNjdlYjAwZDI0MmU0NWQ2NTgzNTE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gu94ZJcZosGQKIe8yjhXu57WlHIWnJ2DJpTO0QrCCcs)

Oh, ok. I think you missed the main message. This is just a note in my message. The main message is to say that I am able to run the code without problem because you said that you were not able to run this on your end. The notes are just to say what environment I am using to run the code, which is JDK 21 (I was able to run it and get results). I also tried your suggestion to add the apache dependency. But when I add it, build failed regarding the unsupported class. I also tried running the code with the apached dependency in JDK 22 to see if your suggestion will work in that environment, but it still did not.

Anyway, the code is running fine on my end. Is it running on your end?

Berichicko commented:
> > You wrote about the error and I wrote you the solution
> > ![image](https://private-user-images.githubusercontent.com/104761482/328571046-ed02b87d-e7ba-464a-8c95-0c3617093798.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTM5MDAsIm5iZiI6MTcxNTA5MzYwMCwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTcxMDQ2LWVkMDJiODdkLWU3YmEtNDY0YS04Yzk1LTBjMzYxNzA5Mzc5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDUzMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMzQ1MjE2ZTU2NmFiNjkwNDE1NjRlMzY1Yjc4Y2ZhM2ZlYjkyNzg4YzMxNjdlYjAwZDI0MmU0NWQ2NTgzNTE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gu94ZJcZosGQKIe8yjhXu57WlHIWnJ2DJpTO0QrCCcs)
> 
> Oh, ok. I think you missed the main message. This is just a note in my message. The main message is to say that I am able to run the code without problem because you said that you were not able to run this on your end. The notes are just to say what environment I am using to run the code, which is JDK 21 (I was able to run it and get results). I also tried your suggestion to add the apache dependency. But when I add it, build failed regarding the unsupported class. I also tried running the code with the apached dependency in JDK 22 to see if your suggestion will work in that environment, but it still did not.
> 
> Anyway, the code is running fine on my end. Is it running on your end?

I understand. Thank you! Yes it works. I was glad to help you!

testproEBF commented:
> > > You wrote about the error and I wrote you the solution
> > > ![image](https://private-user-images.githubusercontent.com/104761482/328571046-ed02b87d-e7ba-464a-8c95-0c3617093798.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTM5MDAsIm5iZiI6MTcxNTA5MzYwMCwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTcxMDQ2LWVkMDJiODdkLWU3YmEtNDY0YS04Yzk1LTBjMzYxNzA5Mzc5OC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNDUzMjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMzQ1MjE2ZTU2NmFiNjkwNDE1NjRlMzY1Yjc4Y2ZhM2ZlYjkyNzg4YzMxNjdlYjAwZDI0MmU0NWQ2NTgzNTE5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Gu94ZJcZosGQKIe8yjhXu57WlHIWnJ2DJpTO0QrCCcs)
> > 
> > 
> > Oh, ok. I think you missed the main message. This is just a note in my message. The main message is to say that I am able to run the code without problem because you said that you were not able to run this on your end. The notes are just to say what environment I am using to run the code, which is JDK 21 (I was able to run it and get results). I also tried your suggestion to add the apache dependency. But when I add it, build failed regarding the unsupported class. I also tried running the code with the apached dependency in JDK 22 to see if your suggestion will work in that environment, but it still did not.
> > Anyway, the code is running fine on my end. Is it running on your end?
> 
> I understand. Thank you! Yes it works. I was glad to help you!

Ok. Just to be clear. It is running on your end. I thought I might need to change my dependencies because you mentioned you are using Java 11. I don't need to do this? You can run and check my work?

Thanks!

Berichicko commented:
> Ok. Just to be clear. It is running on your end. I thought I might need to change my dependencies because you mentioned you are using Java 11. I don't need to do this? You can run and check my work?

Yes, everything is fine, but try the methods I wrote

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/6ab90f6d-0d88-45d2-b0aa-9b738262b273)

![image](https://github.com/testproEBF/A53-QA-Automation-Framework/assets/104761482/c6985bc2-c134-438f-b279-75e52ebf22f8)

testproEBF commented:
> > Ok. Just to be clear. It is running on your end. I thought I might need to change my dependencies because you mentioned you are using Java 11. I don't need to do this? You can run and check my work?
> 
> Yes, everything is fine, but try the methods I wrote
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328587694-6ab90f6d-0d88-45d2-b0aa-9b738262b273.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTY5NjQsIm5iZiI6MTcxNTA5NjY2NCwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTg3Njk0LTZhYjkwZjZkLTBkODgtNDVkMi1iMGFhLTliNzM4MjYyYjI3My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNTQ0MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jOTg4OTViNDE0NmNlNWE3MmRiNmIyOWFhMjQ0ZTE1YmQ5YzJjMzVlMDY5ZGNiNDc2NDZiMTE4ZTY1OWM0NTY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.GWXH8NC-Fxx1HBuNJjMucaPtSTrkw31hKEsPkDwtwgg)
> 
> ![image](https://private-user-images.githubusercontent.com/104761482/328587996-c6985bc2-c134-438f-b279-75e52ebf22f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwOTY5NjQsIm5iZiI6MTcxNTA5NjY2NCwicGF0aCI6Ii8xMDQ3NjE0ODIvMzI4NTg3OTk2LWM2OTg1YmMyLWMxMzQtNDM4Zi1iMjc5LTc1ZTUyZWJmMjJmOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QxNTQ0MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NWYxNzFhMmU5MzMwYWRkMDBiMjA1M2JhODcxYWU3MWVjNmMwZTQ4NDhmOTA3NTUxYTZjMTVjY2VhNmJmYWMyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.pvimaKY7tUrSi__pvQiq2wx9OjKrwkCubcZSvLONtx8)

Great! Yes, I will add the methods you suggested. I will submit the pull request again after. Thank you!

testproEBF committed:
Refactored

Moved some methods and used scenario outline in artist.feature file.
